source/SpriteParameters.h

```diff
@@ -0,0 +1,121 @@
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef SPRITE_PARAMETERS_H_
+#define SPRITE_PARAMETERS_H_
+
+#include "ConditionSet.h"
+#include "Point.h"
+
+#include <map>
+#include <string>
+#include <tuple>
+
+class Sprite;
+
+
+
+// Class holding all of the animation parameters required to animate a sprite.
+class SpriteParameters {
+public:
+
+	// Supported animation transition types from one state to another.
+	enum TransitionType{IMMEDIATE, FINISH, REWIND, NUM_TRANSITIONS};
+
+
+	// A node defining the parameters for a certain trigger sprite.
+	class AnimationParameters {
+	public:
+		// Animation parameters as found in Body.h .
+		// Frames per second.
+		float frameRate = 2. / 60.;
+		// FPS per second, used to increase (or decrease) rate at which animation plays.
+		float rampUpRate = 0.0f;
+		float rampDownRate = 0.0f;
+		// Frame number used to track at which frame the animation starts.
+		float startFrame = 0.f;
+		// Scale of the frame, typically set to 1.
+		float scale = 1.f;
+		// Percentage of animation that needs to play in order for an action to complete.
+		float indicatePercentage = -1.0f;
+		// Delay in the animation starting in number of frames.
+		int delay = 0;
+		// Delay in the transition of one anim to another (e.g FIRING anim to FLYING anim).
+		// In number of frames.
+		int transitionDelay = 0;
+		// The type of transition to perform.
+		TransitionType transitionType = TransitionType::IMMEDIATE;
+		// Used to indicate whether the animation should start at startFrame.
+		bool startAtZero = false;
+		// Used to indicate whether we should randomize the next frame to be played.
+		bool randomize = false;
+		// Used to indicate whether only the startFrame should be randomized.
+		bool randomizeStart = false;
+		// Used to indicate whether the animation should be looped.
+		bool repeat = true;
+		// Used to indicate whether the animation should be rewinded after being played forwards.
+		bool rewind = false;
+		// Used to indicate whether the animation should be entirely played in reverse.
+		bool reverse = false;
+		// Defines whether an animation has to complete in order for a ship to perform an action
+		bool indicateReady = false;
+		// Center of the body.
+		Point center;
+	};
+
+
+public:
+	typedef std::tuple<const Sprite*, SpriteParameters::AnimationParameters, ConditionSet> SpriteDetails;
+	typedef std::map<int, SpriteParameters::SpriteDetails> SpriteMap;
```

Comment:
this really does not need a typedef

Reply:
The typedef was simply to keep the length of the variable declaration down in the class otherwise in the cpp file we would be seeing a lot of large declaration types. I would like to keep it this way for clarity, are there any alternatives to this?

Reply:
You should probably be using [using](https://en.cppreference.com/w/cpp/language/type_alias).

Reply:
if you remove the unnecessary prefixes, `SpriteParameters::` (you don't need these as you are inside this class) and `std::` (you don't need them in the `.cpp` file), they'll become much shorter

Reply:
I have shortened up the names in the latest commit, take a look and let me know if the solution is appropriate!